test/loopback-dl.test.js

```diff
@@ -645,8 +645,28 @@ describe('DataSource define model', function() {
 
     var User = ds.define('User', {});
     assert.deepEqual(User.definition.properties.id,
-      {type: Number, id: 1, generated: true});
+      {type: Number, id: 1, generated: true, updateOnly: true});
+
+    done();
+  });
+
+  it('check forceId and getUpdateOnlyProperties', function(done) {
+    var ds = new DataSource('memory');
+    var Post = ds.define('Post', {
+      title: {type: String, length: 255},
+      date: {type: Date, default: function() {
+        return new Date();
+      }},
+    });
+
+    // check if forceId is added as true in ModelClass's settings[] explicitly, if id a generated (default) and forceId
+    // in from the model is true(unspecified is 'true' which is the default).
+    Post.settings.should.have.property('forceId').eql(true);
 
+    // check if method getUpdateOnlyProperties exist in ModelClass and check if the Post has 'id' in updateOnlyProperties list
+    Post.should.have.property('getUpdateOnlyProperties');
+    var updateOnlyProps = Post.getUpdateOnlyProperties();
+    updateOnlyProps.should.containEql('id');
```

Comment:
I think in this test we know exactly what properties should be marked as `updateOnly`, don't we? In which case I think it's better to assert the exact content of the returned array:

```js
updateOnlyProps.should.eql(['id']);
```